### ISSUE TITLE
check_updates provides incorrect information on RHEL 8

### ISSUE THREAD

**Issue by KaineKaine** [matteocorti/check_updates#40]
**Describe the bug**

On RHEL 8, I configured Nagios with NRPE, and I use the check_updates plugin. When I run it on command line I see there are 0 updates:

/usr/lib64/nagios/plugins/check_updates
CHECK_UPDATES OK - no updates available | total_updates=0;0;0 security_updates=0;0;0

- On the Nagios web interface, it shows incorrectly  that there are 3 security updates available:
Security Updates
CRITICAL 	05-17-2021 13:29:45 	0d 2h 8m 51s 	3/3 	CHECK_UPDATES CRITICAL - 3 security updates available 

**To Reproduce**

Steps to reproduce the behavior.
-  Install Nagios on a RHEL 8 VM
- Install the following check_updates plugin on a RHEL 8 VM: nagios-plugins-check-updates-1.7.10-1.el8.x86_64  
- Configure the following command in commands.cfg:

define command {
        command_name    check_updates
        command_line    $USER1$/check_updates -H $HOSTADDRESS$ -t 120
}

- Define the following service on the Nagios server:
define service {
    use                     generic-service           ; Name of service template to use
     host_name              RHEL8
    service_description     Security Updates
    check_command           check_nrpe!check_updates
}

**Expected behavior**

I see that there are 3 updates on the Nagios web Interface. When I run the command from the client I see 0 updates available. 

**System (please complete the following information):**

 - OS: [e.g. iOS] RHEL 8
 - OS Version: [e.g. 22]  Red Hat Enterprise Linux release 8.3 (Ootpa)
 - check_updates Version: [e.g. 1.19.0] check-updates-1.7.10-1.el8.x86_64 

**Additional context/output**

Add any other context or output (e.g., from `check_updates -d -v`) about the problem here.

 /usr/lib64/nagios/plugins/check_updates -d -v
[DBG] check_updates version: 1.8.0
[DBG] system info:           Linux tst01 4.18.0-240.22.1.el8_3.x86_64 #1 SMP Thu Mar 25 14:36:04 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux
[DBG]                        NAME="Red Hat Enterprise Linux"
VERSION="8.3 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.3"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.3 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.3:GA"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.3
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.3"
Checking a linux system
Running on Red Hat Enterprise Linux release 8.3 (Ootpa)
[DBG] Updater: yum
Using Yum or DNF
running a Linux kernel: 4.18.0-240.22.1
kernel: running = 4.18.0-240.22.1, installed = 4.18.0-240.22.1
[DBG] Running "/bin/dnf check-update --assumeno --errorlevel=0 -q  2>&1"
CHECK_UPDATES OK - no updates available | total_updates=0;0;0 security_updates=0;0;0

**Comment by matteocorti** [comment]
The debugging output is executing version 1.8 while your setup mentions 1.7.10.
Could you please use the latest version and test again?

**Comment by KaineKaine** [comment]
I am using the latest version and I am getting the same results. Attached is the debugging output from today:

 /usr/lib64/nagios/plugins/check_updates -d -v
[DBG] check_updates version: 1.8.0
[DBG] system info:           Linux localhost 4.18.0-240.22.1.el8_3.x86_64 #1 SMP Thu Mar 25 14:36:04 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux
[DBG]                        NAME="Red Hat Enterprise Linux"
VERSION="8.3 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.3"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.3 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.3:GA"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.3
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.3"
Checking a linux system
Running on Red Hat Enterprise Linux release 8.3 (Ootpa)
[DBG] Updater: yum
Using Yum or DNF
running a Linux kernel: 4.18.0-240.22.1
kernel: running = 4.18.0-240.22.1, installed = 4.18.0-240.22.1
[DBG] Running "/bin/dnf check-update --assumeno --errorlevel=0 -q  2>&1"
CHECK_UPDATES OK - no updates available | total_updates=0;0;0 security_updates=0;0;0


-------------------------------------------

Below is the output from Nagios web interface:

localhost
Security Updates
	
	CRITICAL 	05-18-2021 12:25:31 	0d 0h 1m 33s 	4/4 	CHECK_UPDATES CRITICAL - 3 security updates available 

--------------------------------------

This is from the /var/log/nagios/nagios.log file:

[1621355071] SERVICE ALERT: localhost;Security Updates;CRITICAL;SOFT;3;CHECK_UPDATES CRITICAL - 3 security updates available
[1621355133] SERVICE NOTIFICATION: nagiosadmin;localhost;Security Updates;CRITICAL;notify-service-by-email;CHECK_UPDATES CRITICAL - 3 security updates available
[1621355133] SERVICE ALERT: localhost;Security Updates;CRITICAL;HARD;4;CHECK_UPDATES CRITICAL - 3 security updates available

**Comment by matteocorti** [comment]
DNF has a user specific cache. What happens when you execute the plug-in as the same user as the nrpe process?

**Comment by KaineKaine** [comment]
When I execute the plug-in as the same user as the nrpe process I get the incorrect number of packages for the security updates.

**Comment by KaineKaine** [comment]
The nrpe process is run as the nrpe user on my systems. When I ran check_update with the debug it complained about a permission denied on the following file for RHEL 8 only: /var/log/rhsm/rhsm.log

I ran the following command to resolve the issue: setfacl -m u:nrpe:rxw /var/log/rhsm/rhsm.log 
I did that for all files that it complained about. That resolved the issue for RHEL 8. All RHEL 7 systems are working fine without any changes.
